src/domain-engine.js

```diff
@@ -129,7 +129,7 @@ class DomainEngine {
     return next
   }
 
-  dispatch(state: Object, action: Action) {
+  dispatch(state: Object, action: Action): Object {
```

Comment:
I am wondering whether it would be a useful check to do something like https://flow.org/en/docs/types/generics/#toc-function-types-with-generics for this where `dispatch` takes a `state` which is of a generic type and returns the same generic type. Though I could be misunderstanding how that works...orrrrr that may not be useful